### PR TITLE
Update Backup as a Service chapter, rename BaaS -> K8up

### PR DESCRIPTION
**Pull request by bittner**
@Kidswiss and I overhauled the "Backup as a Service" chapter in the docs.

This goes hand in hand with updating the [K8up docs](https://k8up.io/). All detailed descriptions, tutorials and API reference will be contained there.

In order to point to the latest available version we'll need to have a `latest` version available via a URL that doesn't need constant updating. Can that be done easily in the k8up docs, @akosma? – We should wait with merging this PR until that URL is in place.

### CODE REVIEW

**Comment by akosma** [comment]
LGTM

**Review by akosma — APPROVED** [review]

**Comment by bittner** [comment]
> LGTM

The link to the k8up docs won't work, as it points to a hypothetical "latest" version. Hence, I can't merge the PR as is. Also, the latest version of the k8up docs are still pending release.

Is there a chance to add a "latest" route to the k8up docs in due time, or shall I change the link to an unlucky "version 0.1.8" (that would need constant updating)?

**Comment by bittner** [comment]
@tobru @akosma It would help if we could merge this PR.

Can we take a decision on the `latest` URL path for the k8up docs?

1. We need to release the [latest version](https://github.com/vshn/k8up/releases) of the k8up docs.
1. Ideally, we should add a [/docs/latest](https://k8up.io/docs/latest) URL path that we can use here, to avoid having to update deep links like the one included in this PR

As proposed in our internal Discourse, if it's difficult to implement the `/docs/latest` URL path with Antora we can add the related route simply to the Nginx configuration of the related deployment.

**Comment by bittner** [comment]
@tobru Can you help us with [a decision](https://github.com/appuio/docs/pull/74#issuecomment-633483410) here, please?

**Comment by bittner** [comment]
This issue has now surfaced as an internal ticket: APPU-2707

File: k8up.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    We're happy about your feedback!
+
+What is K8up?
+-------------
+
+On APPUiO we've made backing up your data simple. All you need to do is
+create a backup ``Schedule`` object in your APPUiO namespace. This will
+backup the file system data of your PVCs in that namespace.
+
+- Example: `Backup Schedule <https://k8up.io/docs/latest/backup.html>`__ (k8up documentation)
```
Comment (ccremer) [inline k8up.rst]: I would just remove this line. We already have a link to K8up documentation, no need to maintain a link to a specific page in the docs that might change in any K8up release.

**Review by tobru — APPROVED** [review]

**Pull request merged**